### DISCUSSION TITLE
Running a task after clean only if it's not a rebuild

### DISCUSSION THREAD

**Discussion by lawrence-laz** [dotnet/msbuild#7508] [General]
I have a solution with common output directory. Inside this solution I have the following task in the main startup `.csproj` file:
```xml
<Target Name="FullClean" AfterTargets="Clean">
  <RemoveDir Directories="$(OutDir);" />
</Target>
```
This allows to fully clean the output directory when running "Clean" via Visual Studio.
However, this breaks the "Rebuild" option in Visual Studio, as the directory gets cleaned in the middle of a rebuild.

What I want is to be able to run the `<RemoveDir>` task only when "Clean" is invoked, but not when "Rebuild".
I was looking for something like `Condition="'$(IsRebuild)' != 'true'"`, but couldn't find anything.

Is this possible, and if so, how?

**Comment by benvillalobos** [comment]
Rebuild depends on these targets
https://github.com/dotnet/msbuild/blob/ce7523296d7ddb70fb61a68e84bc2ff38e012e35/src/Tasks/Microsoft.Common.CurrentVersion.targets#L956-L961

What you can do is hook into `BeforeRebuild` and set `SomeProperty` to true. Then condition your `FullClean` target to run only if that property is false.
